src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_upscaler.py

```diff
@@ -0,0 +1,484 @@
+                Guidance scale as defined in [Classifier-Free Diffusion Guidance](https://arxiv.org/abs/2207.12598).
+                `guidance_scale` is defined as `w` of equation 2. of [Imagen
+                Paper](https://arxiv.org/pdf/2205.11487.pdf). Guidance scale is enabled by setting `guidance_scale >
+                1`. Higher guidance scale encourages to generate images that are closely linked to the text `prompt`,
+                usually at the expense of lower image quality.
+            generator (`torch.Generator`, *optional*):
+                A [torch generator](https://pytorch.org/docs/stable/generated/torch.Generator.html) to make generation
+                deterministic.
+            output_type (`str`, *optional*, defaults to `"pil"`):
+                The output format of the generated image. Choose between
+                [PIL](https://pillow.readthedocs.io/en/stable/): `PIL.Image.Image` or `np.array`.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] instead of a
+                plain tuple.
+
+        Returns:
+            [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] or `tuple`:
+            [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] if `return_dict` is True, otherwise a `tuple.
+            When returning a tuple, the first element is a list with the generated images, and the second element is a
+            list of `bool`s denoting whether the corresponding generated image likely represents "not-safe-for-work"
+            (nsfw) content, according to the `safety_checker`.
+        """
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(latents, prompt)
+
+        # 2. Define call parameters
+        batch_size, channels, height, width = latents.shape
+        if isinstance(prompt, str):
+            prompt = [prompt] * batch_size
+
+        device = self._execution_device
+        # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
+        # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
+        # corresponds to doing no classifier free guidance.
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        # 3. Prepare timesteps
+        # We take log-linear steps in noise-level from sigma_max to sigma_min
+        # TODO(Pedro) Fix: create the scheduler with the betas instead
+        sigma_min = self.scheduler.sigmas[-2]   # Last one is zero
+        sigma_max = self.scheduler.sigmas[0]
+        # The +1 comes from k-diffusion
+        sigmas = torch.linspace(np.log(sigma_max), np.log(sigma_min), num_inference_steps+1).exp().to(device)
+        # scheduler.sigmas = torch.cat((sigmas, torch.tensor([0.]).to(device)))
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        self.scheduler.sigmas = sigmas
```

Comment:
This is the noise schedule that was used for training. We need to express it in terms of `betas` so we can create diffusers schedulers.